firebase-installations/src/main/java/com/google/firebase/installations/FirebaseInstallations.java

```diff
@@ -273,12 +283,20 @@ private void executeFidRegistration(PersistedFidEntry persistedFidEntry, AwaitLi
     }
```

Comment:
This is only used in one place, how about putting as inline code and you can also remove listener != null check.

Reply:
As discussed offline, we need this check.